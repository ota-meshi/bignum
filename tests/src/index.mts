import chai from "chai";
import { jestSnapshotPlugin } from "mocha-chai-jest-snapshot";
import { setupEngine } from "../../src/index.mjs";

chai.use(jestSnapshotPlugin());
describe("standard tests", () => {
  const e = setupEngine();
  for (const t of [
    test`1 + 2`,
    test`1+2`,
    test`1+-2`,
    test`1.2+0.8`,
    test`1 + ${3}`,
    test`1 + ${2} * 3`,
    test`(1 + ${2}) * 3`,
    test`1 * -(23 % 4)`,
    test`3*3*PI`,
    test`trunc(300*1.08)`,
    test`trunc(320*1.08)`,
  ]) {
    it(t.name, () => {
      chai.expect(e(...t.param)).toMatchSnapshot();
    });
  }
});

function test<OPERAND>(
  template: TemplateStringsArray,
  ...substitutions: OPERAND[]
) {
  let name = "`";
  for (let index = 0; index < template.length; index++) {
    name += template[index];
    if (index < substitutions.length) {
      name += `\${${substitutions[index]}}`;
    }
  }
  name += "`";
  return {
    name,
    param: [template, ...substitutions] as const,
  };
}
