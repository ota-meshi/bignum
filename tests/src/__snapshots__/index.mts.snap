// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`standard tests \`(1 + \${2}) * 3\` 1`] = `9`;

exports[`standard tests \`1 + \${2} * 3\` 1`] = `7`;

exports[`standard tests \`1 + \${3}\` 1`] = `4`;

exports[`standard tests \`1 + 2\` 1`] = `3`;
